### PR TITLE
Merge commands and use an API workflow for apply

### PR DESCRIPTION
**Pull request by Logiraptor**
This PR replaces the tfexec-based apply with an api workflow. It also merges the two current commands into a single command / docker image so that it may be re-used more easily outside of github actions.

### CODE REVIEW

File: docker/action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+runs:
+    using: 'docker'
+    image: 'Dockerfile'
+inputs:
+  working-dir:
+    default: './'
+    description: 'The directory to run the apply in.'
+  dry-run:
+    default: 'false'
+    description: 'Whether to apply the recommendations or just print tf plan output.'
```
Comment (leizor) [inline docker/action.yml:12]: This description no longer applies, right?

File: docker/cmd/adaptive-metrics/main.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	"os"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatalf("missing command, available commands: pull, apply")
+	}
+
+	switch os.Args[1] {
+	case "pull-recommendations":
```
Comment (leizor) [inline docker/cmd/adaptive-metrics/main.go:14]: Should this be just `pull`?

**Review by leizor — APPROVED** [review]
A couple minor things but overall LGTM, and approving to unblock! 👍

File: docker/action.yml (hunk 1 above)
Comment (Logiraptor) [reply docker/action.yml:12]: [aab6cc6](https://github.com/grafana/adaptive-metrics-autoapply/pull/12/commits/aab6cc625d45da60d4c0ca91e748ea52b32054c5)

File: docker/cmd/adaptive-metrics/main.go (hunk 2 above)
Comment (Logiraptor) [reply docker/cmd/adaptive-metrics/main.go]: [aab6cc6](https://github.com/grafana/adaptive-metrics-autoapply/pull/12/commits/aab6cc625d45da60d4c0ca91e748ea52b32054c5)

**Pull request merged**